internal/handlers/base.go

```diff
@@ -0,0 +1,47 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"github.com/go-chi/chi/v5"
+	"github.com/PerfectStepCoder/shorturl/internal/storage"
+)
+
+
+func ShorterURL(storage *storage.Storage, baseURL string) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		originURLbytes, _ := io.ReadAll(req.Body)
```

Comment:
Не стоит забывать закрывать тело ответа после чтения.
 
```suggestion
		originURLbytes, _ := io.ReadAll(req.Body)
		defer func() {
                      if err := res.Body.Close(); err != nil {
                          log.Errorf("could not close response body: %v", err)
                      }
                  }()
```

Reply:
Поправил